### ISSUE TITLE
1.19 release

### ISSUE THREAD

**Issue by keleko34** [EdwinMindcraft/origins-architectury#197]
What would it take to update this to 1.19? is it a great undertaking/amount of work?

**Comment by EdwinMindcraft** [comment]
Shouldn't be too hard, but I still have bugs to fix in 1.18.2 and forge 1.19 hasn't had a RB yet, which means code may change and hooks crash. If you want to try it yourself, [here](https://github.com/EdwinMindcraft/origins-forge) is the build repository, which includes references to all three mods.

**Comment by keleko34** [comment]
so far the only big change I saw was `TextComponent` was removed and replaced with `Component.literal`, also mixin 7 is no longer in snapshot

**Comment by EdwinMindcraft** [comment]
I know mojang basically changed nothing between 1.18.2 and 1.19. All major changes for 1.19 where contained the 1.18.1->1.18.2 release, and it's not the issue. Origins (Apoli more specifically) way to much mixins to have an unstable codebase, and looking at forge 1.19's changelog, there seem to be a *lot* of patch changes this time.
Plus they **finally** implemented a Jar-in-Jar system, which means I can finally stop merging all 4 mods in one, but at the cost of trying to understand a poorly documented system.

**Comment by EdwinMindcraft** [comment]
Right, so I got into the 1.19 port, and it's going to take some time. The largest change is the removal of `IForgeRegistryEntry`, which I was relying on for most of my code. That means I'm moving everything to the vanilla system (i.e. `Holder` or `RegistryEntry` for yarn users) which is:
1. A breaking change
2. A decent amount of work since that means rewriting apoli's PowerContainer and origins' OriginContainer.

TL;DR: Expect some amount of time before 1.19 release.

**Comment by keleko34** [comment]
> Right, so I got into the 1.19 port, and it's going to take some time. The largest change is the removal of `IForgeRegistryEntry`, which I was relying on for most of my code. That means I'm moving everything to the vanilla system (i.e. `Holder` or `RegistryEntry` for yarn users) which is:
> 
> 1. A breaking change
> 2. A decent amount of work since that means rewriting apoli's PowerContainer and origins' OriginContainer.
> 
> TL;DR: Expect some amount of time before 1.19 release.

Dang, ok. I would help but it may take me awhile to figure out the origins codebase.

**Comment by EdwinMindcraft** [comment]
Test build 1 in pre-releases. Haven't even booted the game to test it but hey, there's *technically* a 1.19 version now :P.
I'll fix errors that will ~~inevitably~~ occur later.

**Comment by Master-OfPuppets** [comment]
> Right, so I got into the 1.19 port, and it's going to take some time. The largest change is the removal of `IForgeRegistryEntry`, which I was relying on for most of my code. That means I'm moving everything to the vanilla system (i.e. `Holder` or `RegistryEntry` for yarn users) which is:
> 
> 1. A breaking change
> 2. A decent amount of work since that means rewriting apoli's PowerContainer and origins' OriginContainer.
> 
> TL;DR: Expect some amount of time before 1.19 release.

About how long?

**Comment by EdwinMindcraft** [comment]
Depending on my affinity for working on the mod, up to a week.

**Comment by EdwinMindcraft** [comment]
You can find test build 3 in the [github releases page](https://github.com/EdwinMindcraft/origins-architectury/releases/tag/1.19-tb3). As stated in the first message, test build for now since forge managed to produce a lot of breaking changes in the past few days, so for now it requires Forge 41.0.63.

**Comment by keleko34** [comment]
I will try it out, we actually put on hold the 1.19 build due to forge not having an official release yet. hopefully soon 🤞
